ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/kzg/BlobProofBundle.java

```diff
@@ -0,0 +1,148 @@
+  /**
+   * @param blobType the type of the blob
+   * @param blob the blob being proven.
+   * @param kzgCommitment the KZG commitment for the blob.
+   * @param kzgProof the KZG proof for the blob.
+   * @param versionedHash the versioned hash of the blob.
+   */
+  public BlobProofBundle(
+      final BlobType blobType,
+      final Blob blob,
+      final KZGCommitment kzgCommitment,
+      final List<KZGProof> kzgProof,
+      final VersionedHash versionedHash) {
+    checkArgument(kzgCommitment != null, "kzgCommitment must not be empty");
+    checkArgument(versionedHash != null, "versionedHash must not be empty");
+    checkArgument(blob != null, "blob must not be empty");
+    checkArgument(kzgProof != null, "kzgProof must not be empty");
+    if (blobType == BlobType.KZG_PROOF && kzgProof.size() != 1) {
+      String errorMessage =
+          "Invalid kzgProof size for versionId 0, expected 1 but got " + kzgProof.size();
+      throw new IllegalArgumentException(errorMessage);
+    }
+    if (blobType == BlobType.KZG_CELL_PROOFS
+        && kzgProof.size() != CKZG4844Helper.CELL_PROOFS_PER_BLOB) {
+      String errorMessage =
+          "Invalid kzgProof size for versionId 1, expected "
+              + CKZG4844Helper.CELL_PROOFS_PER_BLOB
+              + " but got "
+              + kzgProof.size();
+      throw new IllegalArgumentException(errorMessage);
+    }
+    this.blobType = blobType;
+    this.blob = blob;
+    this.kzgCommitment = kzgCommitment;
+    this.kzgProof = kzgProof;
+    this.versionedHash = versionedHash;
+    this.blobCells = computeCells(blob, blobType);
+  }
+
+  private Bytes computeCells(final Blob blob, final BlobType blobType) {
+    if (blobType == BlobType.KZG_CELL_PROOFS) {
+      return CKZG4844Helper.computeCells(blob);
+    }
+    return null;
+  }
+
+  public BlobType getBlobType() {
+    return blobType;
+  }
+
+  public Blob getBlob() {
+    return blob;
+  }
+
+  public KZGCommitment getKzgCommitment() {
+    return kzgCommitment;
+  }
+
+  public List<KZGProof> getKzgProof() {
+    return kzgProof;
+  }
+
+  public VersionedHash getVersionedHash() {
+    return versionedHash;
+  }
+
+  public Optional<Bytes> getBlobCellsBytes() {
+    return Optional.ofNullable(blobCells);
+  }
+
+  @Override
+  public boolean equals(final Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (obj == null || obj.getClass() != this.getClass()) {
+      return false;
+    }
+    var that = (BlobProofBundle) obj;
+    return this.blobType == that.blobType
+        && Objects.equals(this.blob, that.blob)
+        && Objects.equals(this.kzgCommitment, that.kzgCommitment)
+        && Objects.equals(this.kzgProof, that.kzgProof)
+        && Objects.equals(this.versionedHash, that.versionedHash);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(blobType, blob, kzgCommitment, kzgProof, versionedHash);
+  }
+
+  @Override
+  public String toString() {
+    return "BlobProofBundle["
```

Comment:
will this toString blow up the logs if it's actually logged? Do we need a short/loggable version?